src/utils/types.ts

```diff
@@ -10,3 +10,18 @@ export type UnionToIntersection<U> = (U extends any ? (k: U) => void : never) ex
 ) => void
   ? I
   : never
+
+export type JSONPrimitive = string | boolean | number | null | undefined
+export type JSONArray = (JSONPrimitive | JSONObject | JSONArray)[]
+export type JSONObject = { [key: string]: JSONPrimitive | JSONArray | JSONObject }
+export type PrettyJSON<T> = T extends true
```

Comment:
I'm afraid I'm not sure what `PrettyJSON` does (I'm just reviewing on Web UI right now). A small comment or unit tests describing its functionality would be appreciated.